fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/exceptionhandlers/MultipleConstraintViolationExceptionMapper.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.http.commons.exceptionhandlers;
+
+import static org.fcrepo.http.commons.domain.RDFMediaType.TEXT_PLAIN_WITH_CHARSET;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import static javax.ws.rs.core.Response.Status.CONFLICT;
+import static javax.ws.rs.core.Response.status;
+
+import javax.servlet.ServletContext;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Link;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriInfo;
+import javax.ws.rs.ext.Provider;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.fcrepo.kernel.api.exception.MultipleConstraintViolationException;
+import org.slf4j.Logger;
+
+/**
+ * Mapper to display all the various constrainedby links and messages.
+ * @author whikloj
+ */
+@Provider
+public class MultipleConstraintViolationExceptionMapper extends
+        ConstraintExceptionMapper<MultipleConstraintViolationException> implements ExceptionDebugLogging {
+
+    private static final Logger LOGGER = getLogger(MultipleConstraintViolationExceptionMapper.class);
+
+    @Context
+    private UriInfo uriInfo;
+
+    @Context
+    private ServletContext context;
+
+    @Override
+    public Response toResponse(final MultipleConstraintViolationException e) {
+        debugException(this, e, LOGGER);
+
+        final String msg = e.getMessage();
+        final Set<String> exceptionClasses = e.getExceptionTypes().stream().map(Exception::getClass)
```

Comment:
Is the reason you iterate the exceptions twice because you're collapsing the exception types?

Reply:
I think I cleaned this up into a single iteration in the next commit 
https://github.com/fcrepo/fcrepo/pull/1779/commits/28eeb9bf650b07194f99b5ac0cbfcacbb3db3165